api/search.js

```diff
@@ -66,14 +69,101 @@ router.post('/recipes', asyncHandler(async (request, response, _next) => {
     body.query.bool.must_not = excludeTerms.map((term) => ({ match: { ingredients: term } }));
   }
 
+  const dietQueryPart = {
+    bool: {
+      must: [],
+    },
+  };
+  diet.forEach((item) => {
+    if (!get(item, 'tags', null)) {
+      dietQueryPart.bool.must.push(
+        {
+          bool: {
+            must: [
+              { match: { tags: item.key } },
+            ],
+          },
+        },
+      );
+    } else {
+      dietQueryPart.bool.must.push(
+        {
+          bool: {
+            should: item.tags.map((tag) => ({ match: { tags: tag } })),
+            minimum_should_match: 1,
+          },
+        },
+      );
+    }
+  });
   if (diet.length > 0) {
-    body.query.bool.must = diet.map((term) => ({ match: { diet: term } }));
+    body.query.bool.must.push(dietQueryPart);
```

Comment:
Another additional bool must, worth to check if you can just do body.query.push(dietQueryPart) because it already contains a "bool.must" part

Reply:
I don't wanna break this, it was so hard to put together, if you want you can edit this when I merge. sowwy